tests/monitor-mode-tests.bats

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bats
+
+source $BATS_TEST_DIRNAME/common.bash
+
+setup_file() {
+	kubectl --context $CLUSTER_CONTEXT delete --wait --ignore-not-found pods --all
+	kubectl --context $CLUSTER_CONTEXT delete --wait --ignore-not-found -n kubewarden clusteradmissionpolicies --all
+	kubectl --context $CLUSTER_CONTEXT wait --for=condition=Ready -n kubewarden pod --all
+}
+
+@test "[Monitor mode end-to-end tests] Install ClusterAdmissionPolicy in monitor mode" {
+	apply_cluster_admission_policy $RESOURCES_DIR/privileged-pod-policy-monitor.yaml
+}
+
+@test "[Monitor mode end-to-end tests] Launch a privileged pod should succeed" {
+	kubectl_apply_should_succeed $RESOURCES_DIR/violate-privileged-pod-policy.yaml
+	default_policy_server_should_have_log_line "policy evaluation (monitor mode)"
+	default_policy_server_should_have_log_line "allowed: false"
+	default_policy_server_should_have_log_line "cannot schedule privileged containers"
+	kubectl_delete $RESOURCES_DIR/violate-privileged-pod-policy.yaml
+}
+
+@test "[Monitor mode end-to-end tests] Transition policy mode from monitor to protect should succeed" {
+	apply_cluster_admission_policy $RESOURCES_DIR/privileged-pod-policy.yaml
+}
```

Comment:
Just to document... I think you should add test to ensure that the `protect` mode is working after a transition from the `monitor` mode. But you can do that after the improvements that you are doing to detect when the policy in the protect mode is enforced. Like you said during our last talk.

Reply:
Exactly, this is how I got into the mess I'm going to tackle with the next controller PR. Thanks :)